Lib/email/_header_value_parser.py

```diff
@@ -2829,6 +2829,17 @@ def _refold_parse_tree(parse_tree, *, policy):
             _fold_mime_parameters(part, lines, maxlen, encoding)
             continue
 
+        if want_encoding and part.token_type == 'addr-spec':
+            # RFC2047 forbids encoded-word in any part of an addr-spec.
+            if charset == 'unknown-8bit':
+                # Non-ASCII addr-spec came from parsed message; leave unchanged.
+                want_encoding = False
+            else:
+                raise ValueError(
```

Comment:
ValueError is correct, but we don't want smtplib caching *all* ValueErrors, so we're going to need a ValueError subclass here.  How about InvalidMailboxError?

This should probably be added to the generator and/or the policy docs.

Reply:
I wasn't sure whether InvalidMailboxError should subclass MessageError or HeaderWriteError, but went with the more general MessageError.